### PR TITLE
[openapi-gen] Lift Endpoint to cover all generation

### PR DESCRIPTION
**Pull request by c-harding**
- Move Endpoint from `temple.generate.language.service.adt` to `temple.generate`
- Move `temple.generate.language.service` to `temple.generate.service`, as it does not share anything with the client code, so nothing else would go in `temple.generate.language`
- Move `Endpoint#verb` to a private method `ServiceGenerator.verb`, as it is specific to service code.

This is in preparation for fixing #107

### CODE REVIEW

File: src/main/scala/temple/generate/service/go/GoServiceGenerator.scala [hunk 1]
```diff
@@ -212,4 +212,12 @@ object GoServiceGenerator extends ServiceGenerator {
...
     )).map { case (path, contents) => path -> (contents + "\n") }
   }
+
+  private def verb(endpoint: Endpoint): String = endpoint match {
+    case Endpoint.ReadAll => "List"
+    case Endpoint.Create  => "Create"
+    case Endpoint.Read    => "Read"
+    case Endpoint.Update  => "Update"
+    case Endpoint.Delete  => "Delete"
+  }
```
Comment (c-harding) [inline src/main/scala/temple/generate/service/go/GoServiceGenerator.scala]: should this instead be placed in `ServiceGenerator`? Is it Go-specific?

Comment (willjvsmith) [reply src/main/scala/temple/generate/service/go/GoServiceGenerator.scala]: 🚚 -> Endpoint

**Review by willjvsmith — APPROVED** [review]

File: src/main/scala/temple/generate/service/go/GoServiceGenerator.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/generate/service/go/GoServiceGenerator.scala]: Don't think so, happy for it to be moved

**Review by jaylees14 — APPROVED** [review]
LGTM

**Pull request merged**